tests/acceptance/features/webUIResharing/reshareUsers.feature

```diff
@@ -98,126 +116,159 @@ Feature: Resharing shared files with different permissions
     Given user "user2" has shared folder "simple-folder" with user "user1" with "read, share, delete" permissions
+    And user "user1" has accepted the share "simple-folder" offered by user "user2"
     And user "user1" has logged in using the webUI
-    When the user shares folder "simple-folder (2)" with user "User Three" as "Advanced permissions" with permissions "share, delete" using the webUI
+    And the user opens folder "Shares" using the webUI
+    When the user shares folder "simple-folder" with user "User Three" as "Advanced permissions" with permissions "share, delete" using the webUI
+    And user "user3" has accepted the share "simple-folder" offered by user "user1"
     And the user re-logs in as "user3" using the webUI
-    And the user opens the share dialog for folder "simple-folder (2)" using the webUI
+    And the user opens folder "Shares" using the webUI
+    And the user opens the share dialog for folder "simple-folder" using the webUI
     Then the current collaborators list should have order "User Two,User Three"
     And user "User Two" should be listed as "Owner" reshared through "User One" in the collaborators list on the webUI
     And user "user3" should have received a share with these details:
     | field       | value               |
     | uid_owner   | user1               |
     | share_with  | user3               |
-    | file_target | /simple-folder (2)  |
+    | file_target | /Shares/simple-folder  |
     | item_type   | folder              |
     | permissions | share, delete, read |
 
   Scenario: User is allowed to reshare a file/folder with the lesser permissions, and check if it is listed for original owner
     Given user "user2" has shared folder "simple-folder" with user "user1" with "read, share, delete" permissions
+    And user "user1" has accepted the share "simple-folder" offered by user "user2"
     And user "user1" has logged in using the webUI
-    When the user shares folder "simple-folder (2)" with user "User Three" as "Advanced permissions" with permissions "delete" using the webUI
+    And the user opens folder "Shares" using the webUI
+    When the user shares folder "simple-folder" with user "User Three" as "Advanced permissions" with permissions "delete" using the webUI
     And the user re-logs in as "user2" using the webUI
     Then user "User One" should be listed as "Advanced permissions" in the collaborators list for folder "simple-folder" on the webUI
     And custom permissions "share, delete" should be set for user "User One" for folder "simple-folder" on the webUI
     And user "User Three" should be listed as "Advanced permissions" in the collaborators list for folder "simple-folder" on the webUI
     And custom permissions "delete" should be set for user "User Three" for folder "simple-folder" on the webUI
+    When user "user3" has accepted the share "simple-folder" offered by user "user1"
```

Comment:
accepts

Reply:
This is an API step. I don't think we should use API steps on When steps for UI tests.
So I figured it would be better to consider this as a Given step for checking the next condition so. So I reused an existing given step.

@phil-davis need your opinion on this.

Reply:
I think I would stay in the format and make a correct `When` step out of it `When user "user3" accepty the share "simple-folder" offered by user "user1" using the API`
and also I suggest to move it further up to the other WHEN steps